### PR TITLE
[Node Configs] Small refactors and clean ups.

### PR DESCRIPTION
**Pull request by JoshLind**
### Description
This PR makes several small refactors and cleanups to the node configs. Specifically, it:
- Removes unused parameters from methods.
- Renames a few methods for consistency and clarity.
- Adds additional comments to the code.
- Unifies a few methods to reduce code duplication and provide better consistency.

Nothing should logically change with this PR. 😄  Also, there's a bunch more cleanups to be done, but I'm delegating that to future PRs.

### Test Plan
Existing test infrastructure.

### CODE REVIEW

File: config/src/config/persistable_config.rs
Comment (davidiw) [inline config/src/config/persistable_config.rs:14]: I don't see the value in this tweak. If we need to better namespace, because we were loading ... other configs, but I can't imagine loading anything other than a "persisted" config. Because we'd generate an ephemeral config. :shrug:

Comment (JoshLind) [reply config/src/config/persistable_config.rs:14]: Yeah, so I'm not super worried about the wording/naming, the issue I'm seeing is that we have a bunch of `load` methods related to the different configs and they aren't super obvious to me 🤦 , e.g.:
- `NodeConfig::load_config()`: calls the persistable config (i.e., load from disk).
- `node_config.load()`: is a wrapper, e.g., that calls `load_config` (above) and `execution_config.load()` (below).
- `execution_config.load()`:  This is a wrapper that loads the execution config from disk and does post-processing.
- `network_config.load()`: This is a function that just does post-processing.
- ... etc.

We have the same issue with all the various `save()` methods 😢 

All I'm wanting here is to somehow identify the method that *actually does the read/write to disk*, so it's obvious that everything else is in memory, hence the explicit `load_persisted_config` and `save_persisted_config`. Does this make sense?

Comment (JoshLind) [reply config/src/config/persistable_config.rs:14]: If it helps, I'm actually adding a new `NodeConfigLoader` object that should just encapsulate all of the loading, verification and post-processing of the various configs (so we'll have a uniform and consistent way of handling configs going forward). Right now, I find the flows very difficult to reason about. This is actually my next PR (once this lands) 🤞

Comment (davidiw) [reply config/src/config/persistable_config.rs:14]: I guess load_config on a type of persistable config feels redundant... I am absolutely nitpicking.

Comment (JoshLind) [reply config/src/config/persistable_config.rs]: Lol, okay -- I've reverted this change. 😄 Feels like a very odd hill to die on, so I won't debate the rename and pray that this will unblock things.

Comment (davidiw) [reply config/src/config/persistable_config.rs]: I have died on smaller and more ridiculous hills.

**Review by davidiw — APPROVED** [review]

**Review by bchocho — APPROVED** [review]

**Pull request merged**
